src/hooks/core/post/usePostHook.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import {Context} from '../../../context';
+import {setMainFeeds, setTopicFeeds} from '../../../context/actions/feeds';
+import {followUserAnon, setFollow, setUnFollow, unfollowUserAnon} from '../../../service/profile';
+import removePrefixTopic from '../../../utils/topics/removePrefixTopic';
+import TopicPageStorage from '../../../utils/storage/custom/topicPageStorage';
+
+const usePostHook = () => {
+  const [profileContext] = (React.useContext(Context) as any).profile;
+  const [feedsContext, feedsContextDispatch] = (React.useContext(Context) as any).feeds;
+  const {myProfile} = profileContext;
+
+  const followUnfollow = async (item: any) => {
+    const user_id = item?.actor?.id;
+    const username = item?.actor?.data?.username;
+    const data = {
+      user_id_follower: myProfile?.user_id,
+      user_id_followed: user_id,
+      username_follower: myProfile?.username,
+      username_followed: username,
+      follow_source: 'feed'
+    };
+    const dataFollowAnon = {
+      follow_source: 'post',
+      post_id: item?.id
+    };
+    const indexFeed = feedsContext?.feeds?.findIndex((feed) => {
+      return feed?.id === item?.id;
+    });
+    const feedData = feedsContext?.feeds[indexFeed];
+    if (feedData?.is_following_target) {
+      if (!feedData?.anon_user_info_color_name) {
+        feedsContext?.feeds.forEach((feed, index) => {
+          if (feed?.actor?.id === user_id) {
+            feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
+          }
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Directly mutating the state is not recommended in React. It can lead to unexpected behavior because React may batch multiple `setState` calls into a single update for performance. Instead, use functional updates provided by `useState` hook or create a new array and then set it as the new state.

```diff
- feedsContext?.feeds.forEach((feed, index) => {
-   if (feed?.actor?.id === user_id) {
-     feedsContext.feeds[index] = {...feedsContext?.feeds[index], is_following_target: false};
-   }
- });
+ const updatedFeeds = feedsContext?.feeds.map(feed => 
+   feed?.actor?.id === user_id ? {...feed, is_following_target: false} : feed
+ );
+ feedsContextDispatch({type: 'UPDATE_FEEDS', payload: updatedFeeds});
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->